### PR TITLE
Add docs for agent 'otel' mode

### PR DESCRIPTION
**Pull request by kilfoyle**
This is a redo of https://github.com/elastic/ingest-docs/pull/1095 reducing the docs that we include in the Fleet & Elastic Agent Guide. Full docs for running Elastic Agent as an OTel collector can be found in the [elastic/opentelemetry repo](https://github.com/elastic/opentelemetry/tree/main).

This updates the F&A Guide with:
 - A brief intro: [Run Elastic Agent as an OTel Collector](https://ingest-docs_bk_1250.docs-preview.app.elstc.co/guide/en/fleet/master/otel-agent.html)
 - The [`elastic-agent otel` command](https://ingest-docs_bk_1250.docs-preview.app.elstc.co/guide/en/fleet/master/elastic-agent-cmd-options.html#elastic-agent-otel-command)

FYI @ycombinator & @strawgate

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/otel-agent.asciidoc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[[otel-agent]]
+= Run {agent} as an OTel Collector
+
+preview::[]
+
+The link:https://opentelemetry.io/docs/collector/[OpenTelemetry Collector] is a vendor-neutral way to receive, process, and export telemetry data. {agent} includes an embedded OTel Collector, enabling you to instrument your applications and infrastructure once, and send data to multiple vendors and backends. 
+
+When you run {agent} in `otel` mode it supports the standard OTel Collector configuration format that defines a set of receivers, processors, exporters, and connectors. Logs, metrics, and traces can be ingested using OpenTelemetry data formats.
+
+For a full overview and steps to configure {agent} in `otel` mode, including a guided onboarding, refer to the link:https://github.com/elastic/opentelemetry/tree/main[Elastic Distributions for OpenTelemetry] repository in GitHub. You can also check the <<elastic-agent-otel-command,`elastic-agent otel`>> in the {fleet} and {agent} Command reference.
```
Comment (kilfoyle) [inline docs/en/ingest-management/elastic-agent/otel-agent.asciidoc:10]:
```suggestion
For a full overview and steps to configure {agent} in `otel` mode, including a guided onboarding, refer to the link:https://github.com/elastic/opentelemetry/tree/main[Elastic Distributions for OpenTelemetry] repository in GitHub. You can also check the <<elastic-agent-otel-command,`elastic-agent otel` command>> in the {fleet} and {agent} Command reference.
```

**Comment by kilfoyle** [comment]
@andrzej-stencel or @michalpristas  would either of you have a bit of time to look over this PR, especially the [elastic-agent otel command](https://ingest-docs_bk_1250.docs-preview.app.elstc.co/guide/en/fleet/master/elastic-agent-cmd-options.html#elastic-agent-otel-command) docs? Thanks.

File: docs/en/ingest-management/commands.asciidoc [hunk 2]
```diff
@@ -886,6 +887,67 @@ elastic-agent install --url=https://fleet-server:8220 \
...
+Run {agent} as an <<otel-agent,OpenTelemetry Collector>>. 
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent otel [flags]
+elastic-agent otel [command]
+kilfoyle marked this conversation as resolved.
```
[the remark this answers is not in the document]
Comment (karenzone) [reply docs/en/ingest-management/commands.asciidoc:905]: I noticed this when I was checking to see if you were ready for final approval. Still happy to approve after all review comments are in.

[the remark this answers is not in the document]
Comment (kilfoyle) [reply docs/en/ingest-management/commands.asciidoc:905]: Oops! How did that get there? Thanks for catching it.

**Review by andrzej-stencel — APPROVED** [review]

**Review by karenzone — APPROVED** [review]
LGTM. :rocket

**Pull request merged**